sdk/monitor/azure-monitor-query/azure/monitor/query/_models.py

```diff
@@ -564,58 +530,104 @@ def _from_generated(cls, generated) -> "MetricAvailability":
         )
 
 
-class MetricAggregationType(str, Enum, metaclass=CaseInsensitiveEnumMeta):
-    """The aggregation type of the metric."""
 
-    NONE = "None"
-    AVERAGE = "Average"
-    COUNT = "Count"
-    MINIMUM = "Minimum"
-    MAXIMUM = "Maximum"
-    TOTAL = "Total"
+class MetricDefinition:  # pylint: disable=too-many-instance-attributes
+    """Metric definition class specifies the metadata for a metric."""
+
+    dimension_required: Optional[bool] = None
+    """Flag to indicate whether the dimension is required."""
+    resource_id: Optional[str] = None
+    """The resource identifier of the resource that emitted the metric."""
+    namespace: Optional[str] = None
+    """The namespace the metric belongs to."""
+    name: Optional[str] = None
+    """The name and the display name of the metric, i.e. it is a localizable string."""
+    unit: Optional[Union[str, MetricUnit]] = None
+    """The unit of the metric. Possible values include "Count", "Bytes", "Seconds", "CountPerSecond",
+    "BytesPerSecond", "Percent", "MilliSeconds", "ByteSeconds", "Unspecified", "Cores", "MilliCores",
+    "NanoCores", "BitsPerSecond"."""
+    primary_aggregation_type: Optional[Union[str, MetricAggregationType]] = None
+    """The primary aggregation type value defining how to use the values for display. Possible values
+    include: "None", "Average", "Count", "Minimum", "Maximum", "Total"."""
```

Comment:
When I generated the documentation with sphinx, I saw the docs get rendered kind of strange with any sentences including a `:`. Might be worth checking that it looks as expected.

Reply:
Great catch! I noticed the same when I generated the docs as it seems to interpret everything preceding the colon as a type. I thought I removed all the colons, but looks like I missed one. Fixed!